### PR TITLE
Add cacheStoragePolicy option

### LINKED ISSUE #72: URLSessionConfiguration.urlCache support
I've noticed that when I use `MockingURLProtocol` URL cache is not used.

given this mock:
```
Mock(url: URL(string: "https://example.com/foo")!,
     dataType: .json,
     statusCode: 200,
     data: [.get: #"{"id": "1"}"#.data(using: .utf8)!],
     additionalHeaders: ["Cache-Control": "public, max-age=31557600, immutable"]
).register()
````

and 

```
configuration.urlCache = URLCache.shared
```

I'd expect response cached (and it does for regular network requests). When the response is mocked though, cache is unused. How do you think is feasible to support cache here?

### PR DESCRIPTION
**Pull request by krzyzanowskim**
This PR adds optional `cacheStoragePolicy`. Defaults to .notAllowed for backward compatibility.

```swift
Mock(url: originalURL, cacheStoragePolicy: .allowed,
     dataType: .json, statusCode: 200,
     data: [.get: MockedData.exampleJSON.data],
     additionalHeaders: ["Cache-Control": "public, max-age=31557600, immutable"]
).register()
```

Resolves https://github.com/WeTransfer/Mocker/issues/72

### CODE REVIEW

**Review by batuhansk — APPROVED** [review]

File: Sources/Mock.swift
Comment (AvdLee) [inline Sources/Mock.swift]: Very, verrrry NIT but, you know

```suggestion
    /// Allow response cache.
```

**Review by AvdLee — APPROVED** [review]
Nice, a simple but very welcome update! 🚀

**Comment by wetransferplatform** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Consider to place some `MARK:` lines for Sources/Mock.swift, which is over 200 lines big.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/4a140de7c1a5d2bf" target="_blank">Bitrise</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Mocker: Executed 22 tests, with 0 failures (0 unexpected) in 1.789 (1.805) seconds</td>
    </tr>
  </tbody>
</table>

## Mocker.framework: Coverage: 88.65
| File | Coverage ||
| --- | --- | --- |
Mock.swift | 84.78% | ✅
MockingURLProtocol.swift | 86.54% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 3d297a4dbd1bd055a2934ed2b0e24856d46063d1
</p>

**Comment by krzyzanowskim** [comment]
will it happen?

**Comment by AvdLee** [comment]
> will it happen?

It's unbelievable. Today I woke up and thought: Let's merge everything! It's a feeling hard to describe but best understandable after pressing the green merge button. 

Amen 🙏

**Pull request merged**
